src/Sulu/Component/Rest/ListBuilder/Doctrine/FieldDescriptor/DoctrineCountDistinctFieldDescriptor.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor;
+
+/**
+ * This class defines the necessary information for a field to resolve it within a Doctrine Query for the ListBuilder.
+ *
+ * @ExclusionPolicy("all")
+ */
+class DoctrineCountDistinctFieldDescriptor extends DoctrineFieldDescriptor
+{
+    public function getSelect()
+    {
+        return 'COUNT(DISTINCT(' . $this->encodeAlias($this->getEntityName()) . '.' . $this->getFieldName() . '))';
```

Comment:
Looks for me like the `DoctrineGroupConcatFieldDescriptor` has a `distinct` flag in its constructor for this kind of query, so maybe it would be better to extend the `DoctrineDistinctFieldDescriptor`

https://github.com/sulu/sulu/blob/release/2.1/src/Sulu/Component/Rest/ListBuilder/Doctrine/FieldDescriptor/DoctrineGroupConcatFieldDescriptor.php#L68

/cc @danrot what do you think?

Reply:
If you are talking about the `DoctrineCountFieldDescriptor` that's ok for me :slightly_smiling_face: